gender_novels/analysis/dependency_parsing.py

```diff
@@ -15,6 +19,69 @@ def load_jars(path_to_jar, path_to_models_jar):
     if not os.path.isfile(path_to_models_jar):
         urllib.request.urlretrieve(url_to_models_jar, path_to_models_jar)
 
+
+def count_gender_subj_obj(tree):
+    """
+    This function takes in an dependency tree for a sentence and counts female and male subject and
+    object occurences
+
+    We have chosen not to include indirect object positions because whether or not they represent
+    passivity (at least, compared to being a direct object) is debatable
+
+    :param tree: An array which is a dependency tree
+    :return: the counts of male and female subject and object occurences as a tuple of 4
+    """
+
+    male_subj_count = male_obj_count = female_subj_count = female_obj_count = 0
+    for triple in tree:
+        if triple[1] == "nsubj" and triple[2][0] == "he":
+            male_subj_count += 1
+        if triple[1] == "dobj" and triple[2][0] == "him":
+            male_obj_count += 1
+        if triple[1] == "nsubj" and triple[2][0] == "she":
+            female_subj_count += 1
+        if triple[1] == "dobj" and triple[2][0] == "her":
+            female_subj_count += 1
+
+    return (male_subj_count, male_obj_count, female_subj_count, female_obj_count)
+
+
+def parse_sentence(sentence):
+    """
+    This function does all sentence parsing (we cannot split this up into separate functions for
+    performance reasons (each additional function will require iterating over the entire tree again)
+    :param sentence: sentence (string) we want to parse
+    :return: the counts of male and female subject and object occurrences as a tuple of 4
+    """
```

Comment:
Here is a sample of how a doc test might look like. You may need to pass the parser as an argument into the function.
```
    >>> path_to_jar = "assets/stanford-parser.jar"
    >>> path_to_models_jar = "assets/stanford-parser-3.9.1-models.jar"
    >>> load_jars(path_to_jar, path_to_models_jar)
    >>> dependency_parser = StanfordDependencyParser(path_to_jar, path_to_models_jar)
    >>> sentence = "Here is a test sentence that we want to parsep"
    >>> parse_sentence(sentence)
    Dependency Tree of the Sentence
```